packages/utils/src/waitFor.ts

```diff
@@ -70,9 +70,13 @@ export function createElapsedTimeTracker({minElapsedTime}: {minElapsedTime: numb
   function elapsedTimeTracker(): boolean {
     const now = Date.now();
     const msSinceLastCall = now - (lastTimeCalled ?? 0);
-    lastTimeCalled = now;
 
-    return msSinceLastCall > minElapsedTime;
+    if (msSinceLastCall > minElapsedTime) {
```

Comment:
I am not in favor of this change in `createElapsedTimeTracker` as it's just to track the total elapsed time without any frequency. 

If there is no other usage then we can rename this to `createTimeFrequencyTracker` or similar to make it clear. 

Or I would better suggest to use the `rateLimiter` here. We have few implementation that we can move to utils and reuse here.